### ISSUE TITLE
Files downloaded from VSO have their filenames altered if they include more that one dot (a.b.fits -> a_b.fits)

### ISSUE THREAD

**Issue by samaloney** [sunpy/sunpy#7450]
### Describe the bug

Files downloaded from the VSO with file names containing more than one dot `a.b.fits` are save to files with altered filenames (`a_b.fits`) which can cause some confusion.

While investigating #6656 I noticed the file names are altered e.g. a url with file name of `20240209120002Uh.fits.fz` will become `20240209120002uh_fits.fz` when downloaded (I know the filename is not taken from the url).

I've tracked it down to the `slugify` function `slugify('20240209120002Uh.fits.fz')` -> `'20240209120002uh_fits.fz'`. I'm not sure if this is really bug with `slugify` or its use in sanitising possible filenames?

L326

https://github.com/sunpy/sunpy/blob/8349d7f8698da737b884faa81b1773ddd1adaf1f/sunpy/net/vso/vso.py#L290-L335

### To Reproduce

VSO seems to be down right now so can't get a full example but this shows the gist

```python
from sunpy.util.net import slugify

slugify('a.b.c.fits')

#'a_b_c.fits'
```

**Comment by ViciousEagle03** [comment]
I think if we specifically pass the parameter of `delim = '.'` , when we call the function `slugify` as `slugify('a.b.c.fits' , delim='.')` , we will get the desired result for the cases where there are multiple `period '.'`.
So do you want the function `slugify` to default to `delim='.'` instead of delim='_'  or only for cases where there are multiple periods in the `FITS` filename?

**Comment by nabobalis** [comment]
We would need to make sure we preserve the filenames with two dots when it comes to file extensions.

**Comment by ViciousEagle03** [comment]
Could you please review my pull request? Thank you.

**Comment by ayshih** [comment]
Why does `slugify()` even replace periods?  Periods are not typically a character that needs to be sanitized.  Can we simply take periods out of the replace list?

I'm not a fan of a fix that attempts to sometimes replace periods, because that invariably means false positives/negatives and maintenance burden.

**Comment by samaloney** [comment]
I think a `slugify` and a `sanitise_filename` function would have different behaviours. It not totally clear what the original intent of `slugify` was and if that's really what is needed here. The example I found was for periods but there maybe other valid filename characters that are also being replace. I Filename sanitisation is as straightforward as it may seem.

**Comment by ayshih** [comment]
Divining the intent from the code, `slugify()` is meant to be an OS-independent sanitization, and as such, some characters are being replaced that do not need to be on some OSs.  We should ponder whether we would like OS-specific sanitization, which has the positive that we can preserve the intended filename on some systems, but has the negative that users on different OSs can have different filenames for the same downloaded file.

**Comment by hasrat17** [comment]
Hi @nabobalis I would like to work on this issue. Can you assign it to me?

**Comment by nabobalis** [comment]
> Hi @nabobalis I would like to work on this issue. Can you assign it to me?

Hello.

We don't assign issues as a general rule.

In addition, this issue already has an open pull request that will close this issue.

I would suggest another issue to tackle.

**Comment by samaloney** [comment]
Closed by #7453
